ui-tests/src/test/java/io/syndesis/qe/steps/customizations/connectors/ApiClientConnectorsSteps.java

```diff
@@ -72,16 +154,41 @@ public void createNewApiConnector(String user, String connectorName, DataTable p
 
         WizardSteps wizardSteps = new WizardSteps();
 
-        List<String> row = asList("file", "swagger/connectors/petstore.json");
-        List<List<String>> dataTable = new ArrayList<>();
-        dataTable.add(row);
-        wizardSteps.uploadSwaggerFile(user, DataTable.create(dataTable));
+        if(sourceDataTable==null) {
+            Assert.fail("Swagger upload type and path not set");
+        }
+        log.info("Setting up upload type and path of the swagger file");
+        wizardSteps.uploadSwaggerFile(user, DataTable.create(sourceDataTable));
+
+        wizardSteps.navigateToNextWizardStep(user, "Review Actions");
+        wizardSteps.navigateToNextWizardStep(user, "SpecifySecurity");
+
+        if(connectorName != null) {
+            log.info("Setting up auth type");
+            wizardSteps.setUpSecurity(user, DataTable.create(securityDataTable));
+            wizardSteps.navigateToNextWizardStep(user, "Review/Edit Connector Details");
+        }
 
-        wizardSteps.navigateToNextWizardStep(user, "Review Swagger Action");
-        wizardSteps.navigateToNextWizardStep(user, "Security");
+        if(connectorName != null) {
+            log.info("Setting up connector name");
+            wizardSteps.setUpConnectorName(user, connectorName);
+        }
+
+        if(description != null) {
+            log.info("Setting up connector name");
+            wizardSteps.setUpDescription(user, description);
+        }
+
+        if(host != null) {
+            log.info("Setting up host");
+            wizardSteps.setUpHost(user, host);
+        }
+
+        if(baseUrl != null) {
```

Comment:
baseUrl is always null - maybe you forgot to add baseUrl parsing?

Reply:
yes, I missed that. Fixed